modules/tinymce/src/core/main/ts/paste/Commands.ts

```diff
@@ -28,6 +28,10 @@ const register = (editor: Editor, pasteFormat: Cell<string>): void => {
     if (value.text) {
       Clipboard.pasteText(editor, value.text);
     }
+
+    if (value.imageEvent) {
```

Comment:
I think this could be refactored to be `value.files` if you separate the `Clipboard.pasteImageData` into `Clipboard.pasteFiles` and `Clipboard.pasteImageDataFromEvent`. Then have the `pasteFiles` take an array of files instead that goes though an image filter as it does when extracting files from the clipboard dataTransfer instances. The the `pasteImageDataFromEvent` would just check what event it is grab the dataTransfer, preventDefault and call the `pasteFiles` function. It would be a bit strange to have a command accepting an event and preventing that event. Commands should be executable outside of events.

Also this needs tests and a changelog since it's an API change. There are multiple tests for this command.

Reply:
Maybe I wasn't good at explaining it I don't think we should make the `mceInsertClipboardContent` command event specific. Instead with a bit of refactoring it could take `html` `text` or `files` since we want to be able to retain file names it needs to be files not blobs. If it accepts an array like of files. Then inside the Clipboard module we make a new function that accepts files instead of the events.

So then you could do this:

```
editor.on('paste', (e) => {
  e.preventDefault();
  editor.execCommand('mceInsertClipboardContent', { files: e.clipboardData.files } );
});
```

Or:
```
editor.on('drop', (e) => {
  e.preventDefault();
  editor.execCommand('mceInsertClipboardContent', { files: e.dataTransfer.files } );
});
```

Or for testing:
editor.execCommand('mceInsertClipboardContent', { files: [
  new File([new Blob([], { type: 'text/html' })], 'test.html')
] } );

Reply:
refactored